### ISSUE TITLE
After payment successfully completed not getting transaction_id or order_id from Instamojo server side

### ISSUE THREAD

**Issue by kiraninfogird** [Instamojo/ios-sdk#8]
I have done the integration of Instamojo with my app by using steps are you given on the website. [https://docs.instamojo.com/v1.1/page/ios-sdk]
But I stuck on one place where I can’t know what should I do so please help me on below point which I can share with you.

func paymentCompletionCallBack() {
     if UserDefaults.standard.value(forKey: "USER-CANCELLED") != nil {
         //Transaction cancelled by user, back button was pressed
     }
     if UserDefaults.standard.value(forKey: "ON-REDIRECT-URL") != nil {
          // Check the payment status with the transaction_id
// HERE THE PROBLEM 
// HOW I CAN FETCH ORDER_ID / TRUNCATION_ID OF USER WHEN PAYMENT DONE
     }
     if UserDefaults.standard.value(forKey: "USER-CANCELLED-ON-VERIFY") != nil {
          //Transaction canceled by user when trying to verify UPI payment
     }
}

  
For above function how I get that value from Instamojo payment server, it will give me the response as 1.

**Comment by UISameer** [comment]
Yes, facing same issue. Can someone from Instamojo team look into it.

**Comment by Shruti2208** [comment]
Any answer to this?

**Comment by Shruti2208** [comment]
> Yes, facing same issue. Can someone from Instamojo team look into it.

Did you find an ans to this?

**Comment by sameerjagtap** [comment]
> > Yes, facing same issue. Can someone from Instamojo team look into it.
> 
> Did you find an ans to this?

No, and no reply from instamojo team, i've sent an email also.

**Comment by UISameer** [comment]
> > Yes, facing same issue. Can someone from Instamojo team look into it.
> 
> Did you find an ans to this?

No, no reply from instamojo team, I've sent an email also.

**Comment by Shruti2208** [comment]
I think i managed to crack it.I made the order a global variable in the class.After successful payment the control reaches the callback method "paymentCompletionCallBack". Try printing the **order.id!**. That should give you the paymentId.

**Comment by UISameer** [comment]
No, that's not the paymentID. paymentID will be something like "MOJO7726005A06926769".
Yo can check the Get payment Details API. "https://docs.instamojo.com/v2/docs/get-payment-details-1"
